plotter/plot_classes/vertex_plot.py

```diff
@@ -0,0 +1,266 @@
+    reorderd based on truth vertex indices.
+    
+    Parameters:
+    ----------
+        true_vi: array of truth vertex indices
+        pred_vi: array of model predicted vertex indices
+        pred_pileup: array of predicted pileup origins
+        pred_fake: array of predicted fake origins
+        pred_prompt: array of predicted prompt origins
+        pred_disp: array of predicted displaced origins
+
+    Returns:
+    -------
+        vi_matrices: list, contains both truth and predicted vertex index plots
+    """
+
+    # initialize list to store vertex index matrices
+    vi_matrices = []
+
+    n = len(true_vi)
+
+    # initialize vertex index (vi) matrices as completely unpaired
+    # note in these matrices: 0 -> track pairs, 1 -> tracks not paired
+    mat_true = np.ones((n,n))
+    mat_pred = np.ones((n,n))
+
+    # create truth vi matrices
+    for i in range(n):
+        # truth vertex: checking if the i^th track is pileup (i.e. not a valid vertex)
+        if true_vi[i] == -2:
+            mat_true[i][i] = 0
+
+        # constructing vertex index relationships for valid tracks
+        else:
+            # check for vi pairs between the i^th and j^th tracks
+            for j in range(n):
+                # checking for matching vertex pairs
+                if true_vi[j] == true_vi[i]:
+                    mat_true[i][j] = 0
+
+    vi_matrices.append(mat_true)
+    
+    # create predicted vi matrix
+    for i, (pu, fk, pr, dp) in enumerate(zip(pred_pileup, pred_fake, pred_prompt, pred_disp)):
+        # determining predicted origin type of track i
+        origin = max(pu, fk, pr, dp)
+
+        # checking if predicted origin is pilup or fake
+        if (origin == pu) or (origin == fk):
+            mat_pred[i][i] = 0
+
+        # checking if predicted origin is prompt
+        elif (origin == pr) or (origin == dp):
+            # check for vi pairs between the i^th and j^th tracks
+            for j in range(n):
+                if pred_vi[j] == pred_vi[i]:
+                    mat_pred[i][j] = 0
+
+    vi_matrices.append(mat_pred)
+
+    # return vi_matrices
+    return vi_matrices
+
+
+class VertexPlotBase(PlotBase):
+    def plot(self):
+        """
+        VertexPlotBase subclass of PlotBase to plot the vertex index matrices for both true
+        and predicted.
+        """
+
+        # required parameters for vertex index plot base. Set in 'style' key in config
+        required_params = {
+            'figsize',
+            'fontsize',
+            'label_fontsize',
+            'dpi',
+            'show_entries',
+            'show_percentages',
+            'text_color_threshold',
+        }
+
+        # filter only the necessary parameters from the config file to plot the vertex matrix
+        filtered_params = {
+            key: value for key, value in self.config.style.items() if key in required_params
+        }
```

Comment:
the idea behind this piece of code was to allow for unpacking `**filtered_params` to pass to the `puma` constructors. If you are not using `puma`, then this is not really needed. You could instead construct a `ConfigDict` from `self.config.style` so that you can access the parameters using `.` syntax, which may be cleaner